### PR TITLE
fix: don't try to do an interaction_check if there isn't a user available to check

### LINKED ISSUE #1877: Paginator View doesn't respond if paginator buttons were added via editing the message
### Summary

This seems to be happening with Paginator.edit() in all cases ive been able to test

### Reproduction Steps

Send a message using e.g. ctx.respond or ctx.send
create a paginator object 
and use await `paginator.edit(message)` to add the paginator view to the message
The buttons appear, but they dont work, this also occurs when using a previous view in the initial message which items from that become unresponsive too.

### Minimal Reproducible Code

```python
async def test_cmd(self, ctx: bridge.BridgeContext):
    embed = utils.minimal_embed("Hello World")
    message = await ctx.respond(embed=embed)
    if isinstance(message, discord.Interaction):
        message = await message.original_response()
    if isinstance(message, (discord.WebhookMessage, discord.PartialMessage)):
        message = self.client.get_message(message.id)
    paginator = pages.Paginator(pages=["message 1", "message 2", "message 3", "message 4"])
    await paginator.edit(message)
```

### Expected Results

The buttons should respond like other views when they are edited, they work fine

### Actual Results

Instead I get a "This Interaction Failed" when clicking any of the buttons

### Intents

all

### System Information

discord/__main__.py:47: DeprecationWarning: VersionInfo.releaselevel is deprecated since version 2.3, consider using release_level instead.
  "- py-cord v{0.major}.{0.minor}.{0.micro}-{0.releaselevel}".format(version_info)
discord/__main__.py:49: DeprecationWarning: VersionInfo.releaselevel is deprecated since version 2.3, consider using release_level instead.
  if version_info.releaselevel != "final":
- Python v3.10.9-final
- py-cord v2.3.2-final
- aiohttp v3.8.3
- system info: Linux 6.1.6-arch1-1 #&#8203;1 SMP PREEMPT_DYNAMIC Sat, 14 Jan 2023 13:09:35 +0000

### Checklist

- [X] I have searched the open issues for duplicates.
- [X] I have shown the entire traceback, if possible.
- [X] I have removed my token from display, if visible.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Revnoplex**
## Summary
Fixes #1877
By default, `user=None` in `Paginator` and `Paginator.edit()`. With `author_check=True` being the default in `Paginator`, this ultimately causes the `interaction_check` of `Paginator` to fail as it doesn't have a user to compare which ultimately means `Paginator.edit()` is broken by default. Disabling the author check if there is no user to compare allows `Paginator.edit()` to work by default.

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -33,6 +33,8 @@ These changes are available on the `master` branch, but have not yet been releas
   documentation. ([#2581](https://github.com/Pycord-Development/pycord/pull/2581))
 - Fixed a possible bug where audio would play too fast at the beginning of audio files.
   ([#2584](https://github.com/Pycord-Development/pycord/pull/2584))
+- Fixed paginator buttons not responding when using `Paginator.edit()` with its default
+  parameters ([#2594](https://github.com/Pycord-Development/pycord/pull/2594))
```
Comment (Dorukyum) [inline CHANGELOG.md:37]: Shortened + added dot
```suggestion
- Fixed paginator not responding when using `Paginator.edit()` with default parameters.
  ([#2594](https://github.com/Pycord-Development/pycord/pull/2594))
```

File: discord/ext/pages/pagination.py [hunk 2]
```diff
@@ -1090,8 +1090,10 @@ async def edit(
 
         if page_content.custom_view:
             self.update_custom_view(page_content.custom_view)
-
-        self.user = user or self.user
+        if user or self.user:
+            self.user = user or self.user
+        else:
+            self.usercheck = False
```
Comment (Dorukyum) [inline discord/ext/pages/pagination.py:1096]: Need to test if this still works in the same logic, but it's cleaner:
```suggestion
        if user:
            self.user = user
        elif not self.user:
            self.usercheck = False
```

Comment (Revnoplex) [reply discord/ext/pages/pagination.py:1096]: I'll test it soon

**Review by Revnoplex — COMMENTED** [review]
What about just doing
```python
self.user = user or self.user
if not self.user:
    self.usercheck = False
```

**Review by Dorukyum — APPROVED** [review]

**Review by plun1331 — APPROVED** [review]

**Pull request merged**